PGM/src/main/java/tc/oc/pgm/polls/commands/PollSubCommands.java

```diff
@@ -0,0 +1,186 @@
+        Player initiator = tc.oc.commons.bukkit.commands.CommandUtils.senderToPlayer(sender);
+        Player player = tc.oc.commons.bukkit.commands.CommandUtils.findOnlinePlayer(args, sender, 0);
+
+        if(player.hasPermission("pgm.poll.kick.exempt") && !initiator.hasPermission("pgm.poll.kick.override")) {
+            throw new TranslatableCommandException("poll.kick.exempt");
+        } else {
+            pollManager.startPoll(pollKickFactory.create(userStore.playerId(tc.oc.commons.bukkit.commands.CommandUtils.senderToPlayer(sender)), player));
+        }
+    }
+
+    @Command(
+        aliases = {"next"},
+        desc = "Start a poll to change the next map.",
+        usage = "[map name]",
+        min = 1,
+        max = -1
+    )
+    @CommandPermissions("poll.next")
+    public List<String> pollNext(CommandContext args, CommandSender sender) throws CommandException {
+        final String mapName = args.argsLength() > 0 ? args.getJoinedStrings(0) : "";
+        if(args.getSuggestionContext() != null) {
+            return CommandUtils.completeMapName(mapName);
+        }
+
+        if(!Config.Poll.enabled()) {
+            throw new TranslatableCommandException("poll.disabled");
+        }
+
+        if(restartManager.isRestartRequested()) {
+            throw new TranslatableCommandException("poll.map.restarting");
+        }
+        if(matchManager.hasMapSet()) {
+            throw new TranslatableCommandException("poll.map.alreadyset");
+        }
+        if(TokenUtil.getUser(sender).maptokens() < 1) {
+            throw new TranslatableCommandException("tokens.map.fail");
+        }
+
+        PGMMap nextMap = CommandUtils.getMap(args.getJoinedStrings(0), sender);
+
+        if(pollBlacklist.isBlacklisted(nextMap) && !sender.hasPermission("poll.next.override")) {
+            throw new TranslatableCommandException("poll.map.notallowed");
+        }
+
+        if(onlinePlayers.count() * 4 / 5 > nextMap.getDocument().max_players() && !sender.hasPermission("poll.next.override")) {
+            throw new TranslatableCommandException("poll.map.toomanyplayers");
+        }
+
+        pollManager.startPoll(pollMapFactory.create(sender, nextMap, userStore.playerId(tc.oc.commons.bukkit.commands.CommandUtils.senderToPlayer(sender))));
+        return null;
+    }
+
+    @Command(
+            aliases = {"mutation", "mt"},
+            desc = "Start a poll to set a mutation",
+            usage = "[mutation name]",
+            min = 1,
+            max = -1
+    )
+    @CommandPermissions("poll.mutation")
+    public List<String> pollMutation(CommandContext args, CommandSender sender) throws CommandException {
+
+        if(args.getSuggestionContext() != null) {
+            return StringUtils.complete(args.getSuggestionContext().getPrefix(), mutationQueue.mutationsAvailable().stream().map(mutation -> mutation.name().toLowerCase()));
+        }
+
+        if(!Config.Poll.enabled()) {
```

Comment:
have a class to hold the configuration values, i.e. a specific PollConfiguration